resources/mappings.js

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/node
+"use strict";
+
+const mappings_source = require("./mappings_source.js"),
+    mappings = new Map(Object.entries(mappings_source));
+
+for (let entry of new Set(Object.values(mappings_source))) {
+    // add clean name -> clean name
+    mappings.set(entry, entry);
+}
+
+module.exports = {
+    map: mappings,
+    // TODO wtf?
+    isItem: (str) => str.indexOf("Item") != -1,
+    isAbility: (str) => str.indexOf("Item") == -1 &&
+        (str.indexOf("Ability") != -1 ||
+        str.indexOf("ABILITY") != -1),
+    isHero: (str) => str.indexOf("*") != -1 && str.indexOf("Item") == -1
```

Comment:
What is hacky about it is that it doesn't return true or false. It returns the index of that string, and if doesn't exist it will return -1. This seems a little hacky but it is the only way, except regex, to find out if a string contain certain substring. I used to insert a tilde before indexOf to force it to be a boolean, until includes come out